IBM/ibm-common-service-operator
# Update backup and restore mongo script to work on ROKS and Z

Includes changes from https://github.com/IBM/ibm-common-service-operator/pull/1270 because they have not been merged in yet and the changes for Z needed some of the changes for ROKS.
Copy paste text from that pr:
>Previously, because the script created it's own storage class, the backup_restore_mongo.sh script did not work on ROKS. The changes in this pr fix that by reusing the storage class the original mongo install already uses instead of creating a new one.
Other changes include changing where the file is located as well as the necessary support files to make airgap process easier. Also updated preload_data.sh to remove any instance of oc and replace it with ${OC}.
Testing:
    - get a roks cluster with common services installed
        - really just need iam and mongo
        - needs LDAP setup
    - run conversion script to split two cp namespaces to use two different cs instances
    - verify cannot login with ldap to the new instance
    - run backup_restore_mongo.sh to restore mongo data into new cs instance
    - log in with LDAP to the new cs instance's cp-console
I was able to get this to work using a SERT cluster

For Z changes:
The exact same problem that the preload script had see https://github.com/IBM/ibm-common-service-operator/pull/1308 for more details. 
Essentially, mongo on Z does not accept `--ssl` flags and will fail a job that tries to use them which both the backup and restore jobs did. These changes are the same made for preload just applied to this different situation. Biggest change is scaling mongo operator, replacing icp-mongodb cm, and restarting mongo pods in both the original and target namespace in this script instead of just the original namespace for the preload script. I have gotten this to run on a SERT Z cluster so I think this should be working.

Testing:
- setup cs on z cluster. Create LDAP 
- run conversion script so now there should be mongo and iam running in both original and target namespace
- wait for pods to come ready
- run backup_restore_mongo.sh --bns <original ns> --rns <target ns> -b -r -c
- verify script completes
- verify can login to target namespace using ldap users from original namespace
